lib/matplotlib/font_manager.py

```diff
@@ -1302,8 +1302,11 @@ def _findfont_cached(self, prop, fontext, directory, fallback_to_default,
             if rebuild_if_missing:
                 _log.info(
                     'findfont: Found a missing font file.  Rebuilding cache.')
-                _rebuild()
-                return fontManager.findfont(
+                new_fm = _load_fontmanager(try_read_cache=False)
```

Comment:
Shouldn't the rebuild/update logic rather be encapsulated in a (private?) method of `FontManager`? Calling an external function to get an object of the same type and copying all it's parameters feels like a design flaw.

Reply:
The design is pretty bad, but I'll leave the refactor to another PR.

Reply:
Then, please add a TODO comment, so that somebody stumbling on this code knows it's just a workaround and not a super-clever intended design.

Reply:
done